Extension/src/LanguageServer/extension.ts

```diff
@@ -248,6 +249,11 @@ export async function activate(): Promise<void> {
         clients.timeTelemetryCollector.setFirstFile(activeEditor.document.uri);
         activeDocument = activeEditor.document;
     }
+
+    if (util.extensionContext) {
+        const tool = vscode.lm.registerTool('cpptools-lmtool-configuration', new CppConfigurationLanguageModelTool(clients));
```

Comment:
The `client` ClientCollection is accessible as a global variable, or using the global `getClients`s, in `extension.ts`.  We do recreate client to recover after a crash, but we appear to reuse the same ClientCollection. I'd recommend using `getClients` as needed, just in case.

Reply:
Thanks, changed to use `getClients()`.